### ISSUE TITLE
Migrate copy functionality to Interactivity API + fetch .md endpoint

### ISSUE THREAD

**Issue by daniel-vacic** [thegrowthproject/wp-llms-txt#3]
## Overview

Replace AJAX-based copy functionality with WordPress Interactivity API, fetching directly from the `.md` endpoint instead of `admin-ajax.php`.

**Related:** See `docs/UPGRADE-PLANS.md` - Plan B
**Depends on:** #2 (Block Supports styling) ✅

## Problem

Current implementation uses legacy patterns:
- `admin-ajax.php` endpoint (slow - full WordPress bootstrap on every request)
- Custom JavaScript with manual event listeners
- Manual state management for button text ("Copy" → "Copying..." → "Copied!")
- Separate `class-ui-buttons.php` for AJAX handler
- `wp_localize_script()` to pass nonce/URL to frontend

## Solution

Use WordPress Interactivity API (WP 6.5+):
- Declarative state management via `data-wp-*` directives
- Fetch directly from existing `.md` endpoint (already cached, no WP bootstrap)
- Reactive UI updates without manual DOM manipulation
- No separate AJAX handler needed

### block.json
- [x] Add `"interactivity": true` to supports
- [x] Change `"viewScript"` to `"viewScriptModule"` (ES modules required)

### view.js (complete rewrite)
- [x] Import `store`, `getContext` from `@wordpress/interactivity`
- [x] Create store with namespace `tgp/copy-button`
- [x] Implement `state.buttonText` getter (reactive based on copyState)
- [x] Implement `state.isLoading` getter
- [x] Implement `state.isDisabled` getter
- [x] Implement `actions.copyMarkdown` generator function
- [x] Fetch from `ctx.mdUrl` (the .md endpoint)
- [x] Use `navigator.clipboard.writeText()` for copy
- [x] Handle loading/success/error states with timeouts
- [x] Remove all legacy IIFE code

### render.php
- [x] Add `data-wp-interactive="tgp/copy-button"` to wrapper
- [x] Add `data-wp-context` with JSON: mdUrl, copyState, label texts
- [x] Add `data-wp-on--click="actions.copyMarkdown"` to copy button
- [x] Add `data-wp-bind--disabled="state.isDisabled"` to copy button
- [x] Add `data-wp-class--is-loading="state.isLoading"` to copy button
- [x] Add `data-wp-text="state.buttonText"` to button text span
- [x] Use `wp_interactivity_data_wp_context()` helper function

### style.css
- [x] Add `.is-loading` class styles (opacity, cursor)
- [x] Add loading spinner animation for icon
- [x] Add `@keyframes tgp-spin` animation

### tgp-llms-txt.php
- [x] Remove `require_once` for `class-ui-buttons.php`
- [x] Remove `new TGP_UI_Buttons()` instantiation
- [x] Remove `wp_localize_script()` call for AJAX
- [x] Update `Requires at least` to 6.5
- [x] Bump version to 1.3.0

### class-ui-buttons.php
- [x] Delete this file entirely

## Acceptance Criteria

- [x] Click "Copy for LLM" triggers `actions.copyMarkdown`
- [x] Button shows "Copying..." during fetch
- [x] Button is disabled during copy operation
- [x] Successful copy shows "Copied!" for 2 seconds
- [x] Failed copy shows "Failed" for 2 seconds
- [x] Button resets to original label after state timeout
- [x] Clipboard contains valid markdown content
- [x] Network tab shows fetch to `.md` endpoint (NOT admin-ajax.php)
- [x] No AJAX requests to `admin-ajax.php`
- [x] No JavaScript errors in console
- [x] Multiple blocks on same page work independently
- [x] "View as Markdown" link still opens .md URL in new tab

## Files Changed

| File | Action |
|------|--------|
| `blocks/copy-button/block.json` | Modified ✅ |
| `blocks/copy-button/view.js` | **Rewritten** ✅ |
| `blocks/copy-button/render.php` | Modified ✅ |
| `blocks/copy-button/style.css` | Modified ✅ |
| `tgp-llms-txt.php` | Modified ✅ |
| `includes/class-ui-buttons.php` | **Deleted** ✅ |

### Interactivity API Directives Used

| Directive | Purpose |
|-----------|---------|
| `data-wp-interactive` | Register block with store namespace |
| `data-wp-context` | Pass server data (mdUrl, labels) to client |
| `data-wp-on--click` | Click handler binding |
| `data-wp-bind--disabled` | Reactive disabled attribute |
| `data-wp-class--is-loading` | Reactive class toggle |
| `data-wp-text` | Reactive text content |

### Generator Function Pattern

The Interactivity API uses generator functions for async operations:

```js
actions: {
  *copyMarkdown() {
    const ctx = getContext();
    ctx.copyState = 'loading';
    
    const response = yield fetch(ctx.mdUrl);
    const markdown = yield response.text();
    yield navigator.clipboard.writeText(markdown);
    
    ctx.copyState = 'success';
  }
}
```

## Breaking Changes

- **Requires WordPress 6.5+** (Interactivity API)
- Removes admin-ajax.php endpoint (no API compatibility needed)
- No backward compatibility per discussion

## Testing Scenarios

1. **Happy path:** Click copy → Loading → Copied! → Reset ✅
2. **Network error:** Disconnect → Click → Loading → Failed → Reset ✅
3. **404 error:** Invalid .md URL → Click → Loading → Failed → Reset ✅
4. **Multiple blocks:** Two blocks on page, copy each independently ✅
5. **Rapid clicks:** Click multiple times quickly, should not double-fetch ✅

---

**Implementation:** PR #6 ✅ Merged to dev
